runtime/util/fltconv.c

```diff
@@ -293,7 +293,7 @@ helperCConvertIntegerToFloat(I_32 src)
 		}
 	
 		if (src < 0) {
-			spfInt = (U_32)(0 - src);
+			spfInt = (U_32)(0 - (U_32)src); /* To avoid busy hang in while loop `fltconv_indexLeadingOne32` cause of integer overflow (Seen in GCC 8+) */
```

Comment:
Would this make more sense as (U_32)0 - (U_32)src ?

Reply:
The comment isn't helpful and makes the line too long: please just remove it.

Reply:
@keithc-ca removed the comment and modified the code

Reply:
Thanks. I suspect that the error fixed here was the original motivation for the comment at [1]: perhaps that comment should be removed now?


[1] https://github.com/eclipse/openj9/blob/master/runtime/util/fltconv.c#L430

Reply:
@keithc-ca That warning message states a similar issue but I'm not sure if its due to signed integer overflow as I haven't tested it on `armhf` architecture. As the warning message doesn't specify anything about GCC 8+ compiler issue i have removed the comment.